bot.js

```diff
@@ -1,197 +1,75 @@
 require("dotenv").config();
 
 let TelegramBot = require("node-telegram-bot-api");
+let game = require("./game.js");
 
 if (!process.env.TELEGRAM_TOKEN) {
   throw new Error("TELEGRAM_TOKEN env variable is missing");
 }
 
 const bot = new TelegramBot(process.env.TELEGRAM_TOKEN, { polling: true });
-let gameStart = false;
-let spentCities = new Set();
-let lastLetter = "";
+const usedKeyWords = ["/start", "начать", "сдаюсь"];
+const commands = {
+  START: /\/start/i,
+  START_GAME: /начать/i,
+  STOP_GAME: /сдаюсь/i
+};
 
 bot.on("polling_error", m => console.log(m));
 
-bot.on("message", msg => {
-  const chatId = msg.chat.id;
+bot.onText(commands.START, msg => {
+  const chatID = msg.chat.id;
 
-  if (msg.text.toLowerCase() == "/start" && !gameStart) {
-    console.log("20 " + chatId + " : " + msg.text);
+  if (!(chatID in game.sessions)) {
     let welcomeMessage =
       "Приветствую тебя в игре Города.\nДля начала новой игры напиши Начать";
-    bot.sendMessage(chatId, welcomeMessage);
-  }
-
-  if (msg.text.toLowerCase() == "сдаюсь" && gameStart) {
-    bot.sendMessage(
-      chatId,
-      "Я выиграл!!! \nГорода которые были названы:\n" +
-        [...spentCities].join(", ")
-    );
-    bot.sendMessage(chatId, "Давай еще сыграем!");
-    session(false);
-  }
-
-  if ((msg.text.toLowerCase() == "начать" && !gameStart) || gameStart) {
-    console.log("27 " + msg.text + " spentCities = " + spentCities.size);
-    startGame(chatId, msg.text);
+    bot.sendMessage(chatID, welcomeMessage);
   }
 });
 
-function startGame(chatId, city) {
-  let cities = [
-    "Архангельск",
-    "Борисполь",
-    "Воронеж",
-    "Гюмри",
-    "Донецк",
-    "Енакиево",
-    "Жуковский",
-    "Запорожье",
-    "Изюм",
-    "Йошкар-Ола",
-    "Киев",
-    "Лондон",
-    "Москва",
-    "Николаев",
-    "Орёл",
-    "Париж",
-    "Ростов",
-    "Селидово",
-    "Трускавец",
-    "Уфа",
-    "Флоренция",
-    "Харьков",
-    "Цюрих",
-    "Челябинск",
-    "Шанхай",
-    "Щёлково",
-    "Ыгдыр",
-    "Электросталь",
-    "Юрмала",
-    "Ялта"
-  ];
-  cities = cities.map(item => item.toLowerCase());
-  city = city.toLowerCase();
-
-  //  1. Отправлен сообщение: Привет начинаем игру
-  if (!gameStart) {
-    session(true);
-    let startMessage =
-      "Начинаем игру. \nЯ называю город, а ты должен в ответ назвать любой город \nначинающийся на последнюю букву моего города.\nГорода не должны повторятся.";
-
-    bot.sendMessage(chatId, startMessage);
+bot.onText(commands.START_GAME, msg => {
+  const chatID = msg.chat.id;
 
-    let randCity = randomCity(cities); //проверить randomCity
-    lastLetter = lastValidLetter(randCity);
-    console.log("88 " + randCity + " буква: " + lastLetter);
+  if (!(chatID in game.sessions)) {
+    game.sessions.makeSession(chatID);
 
-    spentCities.add(randCity);
-    console.log(spentCities);
-
-    bot.sendMessage(chatId, firstSymbToUpperCase(randCity));
-    return;
+    let start = game.start(chatID);
```

Comment:
```suggestion
    let { messages }  = game.start(chatID);
```

Reply:
I did not quite understand this remark. You propose to get rid of the `start` variable, and create a new empty object with `messages `and assign the result of the function to it. The question is, how then will I turn to this object if it does not have a name? I can offer a solution to call a function and directly parse the result of this function ...
```suggestion
    game.start(chatID).messages.forEach(messageItem => bot.sendMessage(chatID, messageItem));
```